File: Breadth-First Search Algorithm/Python/Breadth-First-Search-Algorithm.py
from collections import deque

class Graph:
    def __init__(self, vertices):
        self.vertices = vertices
        self.adj_list = [[] for _ in range(vertices)]

    def add_edge(self, src, dest):
        self.adj_list[src].append(dest)
        self.adj_list[dest].append(src)

    def bfs(self, start_vertex):
        visited = [False] * self.vertices
        queue = deque([start_vertex])

        visited[start_vertex] = True

        while queue:
            current_vertex = queue.popleft()
            print("Visited", current_vertex)

            for adj_vertex in self.adj_list[current_vertex]:
                if not visited[adj_vertex]:
                    visited[adj_vertex] = True
                    queue.append(adj_vertex)

g = Graph(6)

g.add_edge(0, 1)
g.add_edge(0, 2)
g.add_edge(1, 3)
g.add_edge(1, 4)
g.add_edge(2, 4)
g.add_edge(3, 5)
g.add_edge(4, 5)

g.bfs(0)

File: Breadth-First Search Algorithm/Java/Breadth-First-Search-Algorithm.java
import java.util.*;

public class BFS {
    static class Graph {
        int vertices;
        LinkedList<Integer>[] adj_list;

        Graph(int vertices) {
            this.vertices = vertices;
            adj_list = new LinkedList[vertices];

            for (int i = 0; i < vertices; i++) {
                adj_list[i] = new LinkedList<>();
            }
        }

        void addEdge(int src, int dest) {
            adj_list[src].add(dest);
            adj_list[dest].add(src);
        }

        void bfs(int start_vertex) {
            boolean[] visited = new boolean[vertices];
            Queue<Integer> queue = new LinkedList<>();

            visited[start_vertex] = true;
            queue.add(start_vertex);

            while (!queue.isEmpty()) {
                int current_vertex = queue.poll();
                System.out.println("Visited " + current_vertex);

                for (int adj_vertex : adj_list[current_vertex]) {
                    if (!visited[adj_vertex]) {
                        visited[adj_vertex] = true;
                        queue.add(adj_vertex);
                                            }
                }
            }
        }
    }

    public static void main(String[] args) {
        Graph g = new Graph(6);

        g.addEdge(0, 1);
        g.addEdge(0, 2);
        g.addEdge(1, 3);
        g.addEdge(1, 4);
        g.addEdge(2, 4);
        g.addEdge(3, 5);
        g.addEdge(4, 5);

        g.bfs(0);
    }
}
